Lib/test/test_tools/test_msgfmt.py

```diff
@@ -121,10 +125,60 @@ def test_nonexistent_file(self):
         assert_python_failure(msgfmt, 'nonexistent.po')
 
 
+class MultiInputTest(unittest.TestCase):
+    """Tests for the issue https://github.com/python/cpython/issues/79516
+        msgfmt.py accepts multiple input files
+    """
+
+    def test_no_outputfile(self):
+        """Test script without -o option - 1 single file"""
+        with temp_cwd(None):
+            shutil.copy(data_dir / 'file2_fr_lf.po', '.')
+            assert_python_ok(msgfmt, 'file2_fr_lf.po')
+            self.assertTrue(
+                filecmp.cmp(data_dir / 'file2_fr_lf.mo', 'file2_fr_lf.mo'),
+                'Wrong compiled file2_fr_lf.mo')
+
+    def test_both_with_outputfile(self):
+        """Test script with -o option and 2 input files
+
+        The current behaviour is to merge entries having distinct ids
+        and keep last one if the same id occurs in multiple files.
+
+        Here the first file has Windows endings (cflr) while second has
+        Unix endings (lf)
+        """
+        with temp_cwd(None):
+            assert_python_ok(msgfmt, '-o', 'file12.mo',
+                             data_dir / 'file1_fr_crlf.po',
+                             data_dir / 'file2_fr_lf.po')
+            self.assertTrue(
+                filecmp.cmp(data_dir / 'file12_fr.mo', 'file12.mo'),
+                'Wrong compiled file12.mo')
```

Comment:
```suggestion
                filecmp.cmp(data_dir / 'file12_fr.mo', 'file12.mo'))
```
I don't think we need the error messages which are not much more informative than just the `test_both_with_outputfile failed`